app/server_test.go

```diff
@@ -42,61 +43,72 @@ func TestStartServerSuccess(t *testing.T) {
 }
 
 func TestReadReplicaDisabledBasedOnLicense(t *testing.T) {
+	cfg := model.Config{}
+	cfg.SetDefaults()
+	driverName := os.Getenv("MM_SQLSETTINGS_DRIVERNAME")
+	if driverName == "" {
+		driverName = model.DATABASE_DRIVER_POSTGRES
+	}
+	dsn := ""
+	if driverName == model.DATABASE_DRIVER_POSTGRES {
+		dsn = os.Getenv("TEST_DATABASE_POSTGRESQL_DSN")
+	} else {
+		dsn = os.Getenv("TEST_DATABASE_MYSQL_DSN")
+	}
+	cfg.SqlSettings = *storetest.MakeSqlSettings(driverName)
+	if dsn != "" {
+		cfg.SqlSettings.DataSource = &dsn
+	}
+	cfg.SqlSettings.DataSourceReplicas = []string{*cfg.SqlSettings.DataSource}
+	cfg.SqlSettings.DataSourceSearchReplicas = []string{*cfg.SqlSettings.DataSource}
+
 	t.Run("Read Replicas with no License", func(t *testing.T) {
```

Comment:
This tests fails (https://app.circleci.com/pipelines/github/mattermost/enterprise/2617/workflows/6fb78090-61eb-4950-a4dd-cc745cb778d4/jobs/9987/tests) with:
```
=== RUN   TestReadReplicaDisabledBasedOnLicense/Read_Replicas_with_no_License
{"level":"warn","ts":1590572693.4912791,"caller":"app/server.go:210","msg":"Sentry reporting is enabled, but SENTRY_DSN is not set. Disabling reporting."}
{"level":"info","ts":1590572693.4980984,"caller":"utils/i18n.go:83","msg":"Loaded system translations","for locale":"en","from locale":"/go/src/github.com/mattermost/mattermost-server/i18n/en.json"}
{"level":"info","ts":1590572693.4982169,"caller":"app/server_app_adapters.go:58","msg":"Server is initializing..."}
{"level":"debug","ts":1590572693.4989524,"caller":"app/enterprise.go:153","msg":"Loading original SAML library"}
{"level":"debug","ts":1590572693.499073,"caller":"utils/html.go:30","msg":"Parsing server templates","templates_directory":"/tmp/testlib280504788/templates"}
{"level":"info","ts":1590572693.5017102,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572693.5026684,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572703.5028367,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572703.503457,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572713.5036135,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572713.50424,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572723.5043964,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572723.5050757,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572733.5053055,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572733.5059352,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572743.5061433,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572743.5068147,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572753.5069869,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572753.5075903,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572763.5077505,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572763.5083895,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572773.5085483,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572773.5092127,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572783.5093582,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572783.5100355,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572793.510321,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572793.5108433,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572803.510985,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572803.511759,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572813.5317965,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572813.5334811,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572823.5337648,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572823.5344796,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572833.5346403,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572833.5353296,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572843.5354831,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572843.5361638,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572853.5362914,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572853.5367975,"caller":"sqlstore/supplier.go:239","msg":"Failed to ping DB","error":"dial tcp 127.0.0.1:5432: connect: connection refused","retrying in seconds":10}
{"level":"info","ts":1590572863.536942,"caller":"sqlstore/supplier.go:227","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1590572863.5375896,"caller":"sqlstore/supplier.go:235","msg":"Failed to ping DB, server will exit.","error":"dial tcp 127.0.0.1:5432: connect: connection refused"}
FAIL	github.com/mattermost/mattermost-server/v5/app	1110.065s
?   	github.com/mattermost/mattermost-server/v5/app/layer_generators	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/manual.test_load_configuration_defaults_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/manual.test_load_configuration_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_bots_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_call_log_api_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_get_bundle_path_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_get_channels_for_team_for_user_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_get_direct_channel_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_get_profile_image_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_member_channels_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_members_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_search_channels_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_search_posts_in_team_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_search_teams_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_send_mail_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_set_profile_image_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_update_user_active_plugin	[no test files]
?   	github.com/mattermost/mattermost-server/v5/app/plugin_api_tests/test_update_user_status_plugin	[no test files]
```